backup/Makefile

```diff
@@ -2,10 +2,9 @@
 # For a clean tree this is simply short commit hash of HEAD, e.g., "4be01eb".
 # Dirty tree has "-dirty" suffix added, e.g., "4be01eb-dirty".
 PRERELEASE ?= $(shell git describe --match= --always --dirty)
-TIMESTAMP ?= $(shell date +%y%m%d%H%M)
```

Comment:
I guess there was some assumption that we would rebuild the images even without anything changing in the code, and something might get updated, since we don't pin versions for the `pip install` commands?

We'd have to pass the timestamps as external inputs (so they're the same for the `make` runs) if that's really needed.

Reply:
It might be a good idea to use version ranges for the deps actually. Could you create a requirements.txt file with at least the major version pinned?

Reply:
I looked at the deps. TBH I don't really use to pin either pip or setuptools, because these are the build systems. certifi is aimless too, because it's a certificate bundle, it just gets certificate updates it doesn't even use semver. The only thing I would pin here is aws cli.

Reply:
Good points, but a bit unrelated to the push workflow I think. 

We can look at the pinning and simplifications in isolation,  I'd like to avoid changing the image alongside this.

Reply:
Yeah, I'd like to avoid combining changes to the actual image while adding the workflow though. 
So as long as omitting the timestamp isn't a big deal in the meantime, I'd first have the workflow and then look into the pinning and simplifications in isolation.